src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,69 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(todo.userId).then(setUser);
+  }, []);
+
+  const dispatch = useAppDispatch();
+
+  const handleClose = () => {
+    dispatch(currentTodoActions.removeTodo());
+  };
 
-export const TodoModal: React.FC = () => {
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
-
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+      {user ? (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #
+              {todo.id}
+            </div>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={handleClose}
+            />
+          </header>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">fugiat veniam minus</p>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">{todo.title}</p>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
-
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+            <p className="block" data-cy="modal-user">
+              {todo.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
```

Comment:
```suggestion
              {todo.completed 
                ? <strong className="has-text-success">Done</strong>
                : <strong className="has-text-danger">Planned</strong>
              )}
```
I think it will look better